Float.Core/UX/NavigationPageNavigationContext.cs

```diff
@@ -93,13 +94,13 @@ public void PresentPage(Page page, bool animated = true)
         }
 
         /// <inheritdoc />
-        public void DismissPage(bool animated = true)
+        public async Task DismissPage(bool animated = true)
         {
-            DeviceProxy.BeginInvokeOnMainThread(() =>
+            await Device.InvokeOnMainThreadAsync(async () =>
```

Comment:
We would need to continue using the `DeviceProxy` class (see https://github.com/gowithfloat/Float.Core/commit/ec7d9f34c51f7e6f7ebc0f64617032cbd12edbba).